### ISSUE TITLE
Implement additional (classification, regression and ranking) metrics

### ISSUE THREAD

**Issue by inejc** [picnicml/doddle-model#12]
Metrics are means of ML algorithms evaluation. See [this](https://github.com/picnicml/doddle-model/tree/master/src/main/scala/io/picnicml/doddlemodel/metrics) for existing implementations.

**Comment by dandxy89** [comment]
Also see - [Sklearn Metric](http://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) and [Caret](https://topepo.github.io/caret/measuring-performance.html)

**Comment by inejc** [comment]
Closing this in favor of more specific issues (e.g. Implement the `zero-one classification` loss).
